apps/api/src/app/workflows-v2/customTypes.ts

```diff
@@ -0,0 +1 @@
+export type RequiredProp<T, K extends keyof T> = T & { [P in K]-?: T[P] };
```

Comment:
<!-- Chores are simple tasks that must be done before the subject can be “officially” accepted. Usually, these comments reference some common process. Try to leave a link to the process description so that the reader knows how to resolve the chore.  -->
☑ **chore:** Please add a typedoc for generic utility types, describing the action on `T` and `K` that the generic performs.

Reply:
will do